File: backend/app/schemas/trading.py
from __future__ import annotations

from pydantic import BaseModel, Field

from app.schemas.execution import GeoblockStatus
from app.schemas.wallet import WalletReadinessResponse


class TradingReadinessResponse(BaseModel):
    wallet: WalletReadinessResponse
    geoblock: GeoblockStatus
    paper_trading_enabled: bool
    bot_running: bool
    user_trading_enabled: bool
    env_trading_enabled: bool
    real_trading_confirmation_enabled: bool
    redeem_enabled: bool
    redeem_dry_run: bool
    trading_enabled: bool
    kill_switch_active: bool
    real_order_dry_run: bool
    wallet_configured: bool
    api_credentials_configured: bool
    sdk_import_ok: bool
    polygon_rpc_configured: bool
    collateral_token_configured: bool
    conditional_tokens_contract_configured: bool
    wallet_redeem_flow_supported: bool
    wallet_redeem_flow_blocking_reason: str | None = None
    official_resolution_client_available: bool
    trading_ready: bool
    paper_trading_ready: bool
    dry_run_trading_ready: bool
    real_trading_ready: bool
    real_trading_available: bool
    blocking_reasons: list[str] = Field(default_factory=list)
    real_trading_blocking_reasons: list[str] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)


class EnableTradingRequest(BaseModel):
    confirm_phrase: str


class TradingStatusResponse(BaseModel):
    trading_enabled: bool
    kill_switch_active: bool
    real_order_dry_run: bool
    mode: str

File: frontend/types/trading.ts
import type { WalletReadiness } from "@/types/wallet";

export type GeoblockStatus = {
  blocked: boolean;
  checked: boolean;
  raw_response: Record<string, unknown>;
};

export type TradingReadiness = {
  wallet: WalletReadiness;
  geoblock: GeoblockStatus;
  paper_trading_enabled: boolean;
  trading_enabled: boolean;
  kill_switch_active: boolean;
  real_order_dry_run: boolean;
  trading_ready: boolean;
  paper_trading_ready: boolean;
  dry_run_trading_ready: boolean;
  real_trading_ready: boolean;
  real_trading_available: boolean;
  blocking_reasons: string[];
  real_trading_blocking_reasons: string[];
  warnings: string[];
};

export type TradingStatus = {
  trading_enabled: boolean;
  kill_switch_active: boolean;
  real_order_dry_run: boolean;
  mode: "dry_run" | "real" | string;
};
